nginx_stage/lib/nginx_stage/generators/pun_config_generator.rb

```diff
@@ -95,6 +95,27 @@ class PunConfigGenerator < Generator
       template "pun.conf.erb", config_path
     end
 
+    # Read standard in, expecting FOO=BAR and set the environment variables
+    # ENV['FOO']=BAR for each line of standard input.
+    add_hook :read_stdin do
+      unless STDIN.tty?
+        STDIN.each_line do |line|
+          key_values = line.split('=')
+          ENV[key_values[0]] = key_values[1] if key_values.size[2]
+        end
+      end
+    end
```

Comment:
Instead of reading stdin to set env vars, why not just pass stdin straight to the pre hook?

Reply:
I just figured parsing in Ruby would be easier, than say bash. Also, we may have multiple hooks. So the pun_pre_hook may be some wrapper that calls multiple scripts.

Reply:
The hook scripts working with env vars is probably easier to work with in hook scripts. But instead of nginx_stage modifying its own environment in "read_stdin" and then executing exec_pre_hook, I'd combine the two, passing the env as an argument i.e. 

```
add_hook :exec_pre_hook do
  if NginxStage.pun_pre_hook
    args = ["--user", user]
    env = {}
    unless STDIN.tty?
      STDIN.each_line do |line|

    Open3.capture2e(env, NginxStage.pun_pre_hook, *args)
  end
end
```

Reply:
:+1: